### ISSUE TITLE
[BUG] Bot stuck on "cancelling the limit order sell" in KuCoin

### ISSUE THREAD

**Issue by masonyoungblood** [CoinAlpha/hummingbot#1804]
Sometimes when order_refresh_tolerance_pct is triggered and the bot tries to replace the buy and sell orders, the sell order remains and the bot repeatedly outputs "cancelling the limit order sell" and never creates new orders. I'm working in paper trading mode on Kucoin, using single orders. The bug occurs with various values for order_refresh_time, order_expiration_time, and filled_order_delay so I don't think it's related to those parameters. I've attached a screenshot of what this looks like.

![Screenshot from 2020-05-23 18-38-09](https://user-images.githubusercontent.com/43799018/82741901-5134ff80-9d25-11ea-9292-deaabfee33d6.png)

**Describe the bug**
// A clear and concise description of what the bug is.

**Steps To Reproduce**
// A concise description of the **steps to reproduce the buggy behavior**:

Here are the steps to reproduce the issue (see attachments in the section below):
1. Set up your configuration file with these parameters: ...
2. Start the bot using this command: ...
...

**Screenshots**
// If applicable, add screenshots to help explain your problem.

**Release version**
// Include your bot's version number (Can be found at the upper left corner of your CLI).

**Attachments**
// Please attach your config file and log file to help us debug. It would be difficult for us to help you without those! 

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**

**Comment by devoger** [comment]
I reproduced the problem in Kucoin and Binance using Paper trade >ON

Binance pair tested: DOCK/BTC
Kucoin Pair Tested: ETN/ BTC and ETH/USDT
Order_refresh_time: 20
Order_refresh_tolerance_pct: 0.1
Filled_order_delay: 10
Order_optimization_enabled - yes

The problem occurs when order_refresh_tolerance is triggered, the bot gets stuck cancelling the limit order sell. The problem occurs when order_optimization is active.

Kucoin ETN/BTC
Here is the configuration:
![etn:config](https://user-images.githubusercontent.com/8411294/82747260-c8bc6a80-9dc9-11ea-9d14-6f7c54e5c5dc.png)


![status-etn](https://user-images.githubusercontent.com/8411294/82747265-d1ad3c00-9dc9-11ea-8efd-fe28d8e7cafb.png)


Binance DOCK/BTC

I started to test with order_optimization_enabled: NO, it was working just fine. I decided to turn it on, and the problem started. I attached the logs.

Configuration
![dock-cnfig](https://user-images.githubusercontent.com/8411294/82747308-32d50f80-9dca-11ea-8883-9acca6c02006.png)


![config-status](https://user-images.githubusercontent.com/8411294/82747312-3799c380-9dca-11ea-890f-3a2fa893d085.png)

**Comment by Clint-G13** [comment]
We had this before on Binance and affects development version only however this was fixed when 0.27.0 was released. Made a test on both KuCoin and Binance (Optimization = enabled) and been running for 3 hours however, I did not see the behavior come up again.
KuCoin using Windows Binary:
![kucoin with optimization enabled](https://user-images.githubusercontent.com/57189990/82857370-612a2a80-9f43-11ea-857e-4590e8804251.PNG)

Binance using master and development branch via Source:
![DOCK-BTC](https://user-images.githubusercontent.com/57189990/82857372-64251b00-9f43-11ea-8967-4b5d284f6c06.PNG)

**Comment by dennisocana** [comment]
@vic-en ran into the same issue. It appears the bug this is caused by websocket connection issues and not reconnecting.

Log file : [kucoin_stuck_canceling_orders.log](https://github.com/CoinAlpha/hummingbot/files/4774227/kucoin_stuck_canceling_orders.log)

```
2020-06-10 01:42:25,192 - 10 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: Error fetching data from https://api.kucoin.com/api/v1/accounts. HTTP status is 504.
Traceback (most recent call last):
  File "/hummingbot/core/utils/async_utils.py", line 9, in safe_wrapper
    return await c
  File "network_iterator.pyx", line 107, in _check_network_loop
  File "kucoin_market.pyx", line 210, in start_network
  File "kucoin_market.pyx", line 287, in _update_account_id
  File "kucoin_market.pyx", line 279, in _api_request
OSError: Error fetching data from https://api.kucoin.com/api/v1/accounts. HTTP status is 504.
```

```
2020-06-10 01:42:25,311 - 10 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: Error fetching Kucoin websocket connection data.HTTP status is 504.
Traceback (most recent call last):
  File "/hummingbot/core/utils/async_utils.py", line 9, in safe_wrapper
    return await c
  File "/hummingbot/market/kucoin/kucoin_api_order_book_data_source.py", line 198, in listen_for_trades
    websocket_data: Dict[str, Any] = await self.ws_connect_data()
  File "/hummingbot/market/kucoin/kucoin_api_order_book_data_source.py", line 192, in ws_connect_data
    raise IOError(f"Error fetching Kucoin websocket connection data."
OSError: Error fetching Kucoin websocket connection data.HTTP status is 504.
```
